src/main/java/com/epam/aidial/core/security/IdentityProvider.java

```diff
@@ -94,7 +105,7 @@ private List<String> extractUserRoles(DecodedJWT token) {
             return token.getClaim(rolePath[0]).asList(String.class);
         }
         Map<String, Object> claim = token.getClaim(rolePath[0]).asMap();
-        for (int i = 1; i < rolePath.length; i++) {
+        for (int i = 1; claim != null && i < rolePath.length; i++) {
```

Comment:
`claim != null` - this check makes sense only for line `Map<String, Object> claim = token.getClaim(rolePath[0]).asMap();`
we can check it before loop and return result immediately, but this is a personal preference

Reply:
fixed